### ISSUE TITLE
Android 12 support

### ISSUE THREAD

**Issue by man4567890** [DeviceFarmer/stf#313]
Hi,
Does the latest release support android 12? SDK-31? , I'm running into errors
```
2021-08-25T05:44:59.196Z INF/device:resources:minicap 4694 [17061FDEE00099] Installing "/usr/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/arm64-v8a/bin/minicap" as "/data/local/tmp/minicap"
2021-08-25T05:44:59.196Z INF/device:resources:minicap 4694 [17061FDEE00099] Installing "/usr/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/arm64-v8a/lib/android-31/minicap.so" as "/data/local/tmp/minicap.so"
2021-08-25T05:44:59.492Z INF/device:resources:service 4694 [17061FDEE00099] Checking whether we need to install STFService
2021-08-25T05:44:59.557Z INF/device:resources:service 4694 [17061FDEE00099] Running version check
2021-08-25T05:45:00.632Z INF/device:resources:service 4694 [17061FDEE00099] STFService up to date
2021-08-25T05:45:00.634Z INF/device:plugins:service 4694 [17061FDEE00099] Launching agent
2021-08-25T05:45:01.312Z INF/device:plugins:service 4694 [17061FDEE00099] Agent says: "Starting minitouch agent"
2021-08-25T05:45:01.312Z INF/device:plugins:service 4694 [17061FDEE00099] Agent says: "Starting minitouch agent"
2021-08-25T05:45:11.130Z FTL/device 4694 [17061FDEE00099] Setup had an error TimeoutError: operation timed out
    at afterTimeout (/usr/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/timers.js:11:15)
    at Timeout.timeoutTimeout [as _onTimeout] (/usr/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/timers.js:53:9)
    at ontimeout (timers.js:498:11)
    at tryOnTimeout (timers.js:323:5)
    at Timer.listOnTimeout (timers.js:290:5)
2021-08-25T05:45:11.131Z FTL/util:lifecycle 4694 [17061FDEE00099] Shutting down due to fatal error
2021-08-25T05:45:11.144Z INF/provider 32128 [*] Cleaning up device worker "17061FDEE00099"

```
Pixel 6 pro

Could you please confirm if Android 12 is supported? so that I have a clearer idea where the problem might be
Thanks

**Comment by man4567890** [comment]
Got it to work after upgrading to 3. 6.1

**Comment by JunaidBabu** [comment]
How did you get **android-31/minicap.so** ? I'm getting the following error

```
2021-11-09T08:36:11.402Z INF/device:support:sdk 29399 [6DD10C9C] Supports SDK 31
2021-11-09T08:36:11.402Z INF/device:support:abi 29399 [6D007C9C] Supports ABIs armeabi-v7a, armeabi
2021-11-09T08:36:11.403Z FTL/device 29399 [6D007C9C] Setup had an error Error: At least one of these paths should exist: 
/home/devnet/.node/lib/node_modules/@devicefarmer/stf/node_modules/@devicefarmer/minicap-prebuilt/prebuilt/armeabi-v7a/lib/android-31/minicap.so, 
/home/devnet/.node/lib/node_modules/@devicefarmer/stf/node_modules/@devicefarmer/minicap-prebuilt/prebuilt/armeabi-v7a/lib/android-31/minicap.so, 
/home/devnet/.node/lib/node_modules/@devicefarmer/stf/node_modules/@devicefarmer/minicap-prebuilt/prebuilt/armeabi/lib/android-31/minicap.so, 
/home/devnet/.node/lib/node_modules/@devicefarmer/stf/node_modules/@devicefarmer/minicap-prebuilt/prebuilt/armeabi/lib/android-31/minicap.so
```

**Comment by man4567890** [comment]
looks like your directory does not exist /home/devnet/.node/lib/node_modules/@devicefarmer/stf/node_modules/@devicefarmer/minicap-prebuilt/prebuilt/armeabi-v7a/lib/android-31/ , create the directory and copy the minicap.so from .../android-30 , sometimes that works

**Comment by ardaozerten** [comment]
Hi, I am facing exact same issue with Pixel 6 Android 12 device. I have updated my stf version to v3.6.1 and copied and pasted minicap.so directories in armeabi-v7a for android-31. Still receiving same timeout error after "Starting minitouch agent". 

I thought if there is a setting might be the cause of this problem, so I tried to give privileges to STFService.apk installed on the device, still not working.

Is there still someone to help out with this problem?

`2021-08-25T05:44:59.196Z INF/device:resources:minicap 4694 [17061FDEE00099] Installing "/usr/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/arm64-v8a/bin/minicap" as "/data/local/tmp/minicap"
2021-08-25T05:44:59.196Z INF/device:resources:minicap 4694 [17061FDEE00099] Installing "/usr/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/arm64-v8a/lib/android-31/minicap.so" as "/data/local/tmp/minicap.so"
2021-08-25T05:44:59.492Z INF/device:resources:service 4694 [17061FDEE00099] Checking whether we need to install STFService
2021-08-25T05:44:59.557Z INF/device:resources:service 4694 [17061FDEE00099] Running version check
2021-08-25T05:45:00.632Z INF/device:resources:service 4694 [17061FDEE00099] STFService up to date
2021-08-25T05:45:00.634Z INF/device:plugins:service 4694 [17061FDEE00099] Launching agent
2021-08-25T05:45:01.312Z INF/device:plugins:service 4694 [17061FDEE00099] Agent says: "Starting minitouch agent"
2021-08-25T05:45:01.312Z INF/device:plugins:service 4694 [17061FDEE00099] Agent says: "Starting minitouch agent"
2021-08-25T05:45:11.130Z FTL/device 4694 [17061FDEE00099] Setup had an error TimeoutError: operation timed out
    at afterTimeout (/usr/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/timers.js:11:15)
    at Timeout.timeoutTimeout [as _onTimeout] (/usr/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/timers.js:53:9)
    at ontimeout (timers.js:498:11)
    at tryOnTimeout (timers.js:323:5)
    at Timer.listOnTimeout (timers.js:290:5)
2021-08-25T05:45:11.131Z FTL/util:lifecycle 4694 [17061FDEE00099] Shutting down due to fatal error
2021-08-25T05:45:11.144Z INF/provider 32128 [*] Cleaning up device worker "17061FDEE00099"`
